hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -25,30 +301,461 @@ The following table lists the main parameters used throughout the OTP controller
+
+#### CSRNG Interface
+
+The entropy request interface that talks to CSRNG in order to fetch fresh entropy for ephemeral SRAM scrambling key derivation and the LFSR counters for background checks.
+It is comprised of the `otp_csrng_o` and `otp_csrng_i` signals and follows a req / ack protocol.
+
+#### Power Manager Interfaces
+
+There are two separate interfaces that connect the OTP controller with the power manager.
+
+First, there is an initialization request interface `pwr_otp_init_i` / `pwr_otp_init_o` that follows a req / ack protocol.
+The power manager asserts `pwr_otp_init_i` in order to signal to the OTP controller that it can start initialization, and the OTP controller signals completion of the initialization sequence by pulsing `pwr_otp_init_o` high.
+
+Second, the OTP controller always outputs its state to the power manager via `otp_pwr_state_o`, such that the power manager can determine whether a shutdown needs to be aborted due to an ongoing OTP operation.
+
+#### Life Cycle Interface
+
+**TODO: need expand this more**
+The token counters are maintained in the OTP.
+To ensure the security of token limits cannot be bypassed, each request for a conditional transition FIRST increments the token count, and THEN checks for the validity of the token.
+
+#### Interface to Key Manager
+
+The interface to the key manager is a simple struct that outputs the CREATOR_ROOT_KEY_SHARE0 and CREATOR_ROOT_KEY_SHARE1 keys via `otp_keymgr_key_o` if these secrets have been provisioned and locked (via CREATOR_KEY_LOCK).
+Otherwise, this signal is tied to all-zeroes.
+
+#### Interfaces to Flash
+
+The interface to the flash scrambling device is similar to the key manager interface and outputs the FLASH_DATA_KEY and FLASH_ADDR_KEY via the `otp_flash_key_o` signal, if the secrets have been provisioned and locked (via FLASH_KEYS_LOCK).
+Otherwise, this signal is tied to all-zeroes.
+
+#### Interfaces to SRAM Scramblers
+
+The interfaces to the FLASH and SRAM scrambling devices follow a req / ack protocol, where the scrambling device first requests a new ephemeral key by asserting the request channel (e.g. `otp_ram_main_key_i`).
+The OTP controller then fetches entropy from CSRNG and derives an ephemeral key using the SRAM_DATA_KEY and the PRESENT scrambling data path as described further below.
+Finally, the OTP controller returns a fresh ephemeral key via the `otp_ram_main_key_o` struct, which completest the req / ack handshake.
+
+Note that this mechanism always works - even if the SRAM_DATA_KEY has not yet been provisioned and locked (via SRAM_KEY_LOCK).
+If SRAM_DATA_KEY has not been locked, an all-zeros value is used to derive the ephemeral key.
 
 
 ## Design Details
 
-# Programmers Guide
+### Block Diagram
+
+**TODO: revamp the diagram below and add more details**
+The following is a high-level block diagram that illustrates everything that has been discussed.
+
+![OTP Controller Block Diagram](otp_controller_block_diagram.svg)
```

Comment:
image source here for reference (WIP):
https://docs.google.com/drawings/d/1nQ7fq3NLit5xozO9MUplAmi6wZ3r0ncHdw1xNZpDe1Y